### PR TITLE
Enable multi-key sorting of a DataFrame

### PR DESCRIPTION
**Pull request by JovanVeljanoski**
- [x] Enable multi-key sorting of a DataFrame
- [x] Add unit-test
- [x] Update docstring to account for the changes.

### CODE REVIEW

File: packages/vaex-core/vaex/dataframe.py [hunk 1]
```diff
@@ -3954,8 +3956,14 @@ def sort(self, by, ascending=True, kind='quicksort'):
         :param str kind: kind of algorithm to use (passed to numpy.argsort)
         '''
         self = self.trim()
-        values = self.evaluate(by)
-        indices = np.argsort(values, kind=kind)
+        if not isinstance(by, list):
+            values = self.evaluate(by)
+            indices = np.argsort(values, kind=kind)
+        if isinstance(by, list):
```
Comment (maartenbreddels) [inline packages/vaex-core/vaex/dataframe.py]:
```suggestion
        if isinstance(by, (list, tuple)):
```
Is more standard

File: packages/vaex-core/vaex/dataframe.py [hunk 2]
```diff
@@ -3954,8 +3956,14 @@ def sort(self, by, ascending=True, kind='quicksort'):
...
+            by = _ensure_strings_from_expressions(by)[::-1]
+            by = ', '.join(by)
```
Comment (maartenbreddels) [inline packages/vaex-core/vaex/dataframe.py]: This is actually not supported, I'm working on a parallel evalulate that takes a list of expressions. I'm ok with merging this, but we need to put a reminder
```suggestion
            by = ', '.join(by)  # TODO: this is using an unsupported features, when evaluate takes a list of expressions, we should change this.
```

File: tests/sort_test.py [hunk 3]
```diff
@@ -26,3 +26,19 @@ def test_sort_filtered():
...
+
+def test_sort_multikey():
+    x = np.array([5, 3, 1, 1, 5])
+    y = np.array([0, 3, 4, 2, 1])
+    z = np.array(['dog', 'cat', 'cat', 'dog', 'mouse'])
+    df = vaex.from_arrays(x=x, y=y, z=z)
+
+    # Case 1: numeric keys
+    df_sorted_1 = df.sort(by=['x', 'y'])
+    assert df_sorted_1.values.tolist() == [[1, 2], [1, 4], [3, 3], [5, 0], [5, 1]]
```
Comment (maartenbreddels) [inline tests/sort_test.py]: You are assuming that the order of the columns is x, then y. This is not supported before Python3.6. Use from_items to ensure order. Of simply use df_sorted_1.x.tolist() and y separate

File: packages/vaex-core/vaex/dataframe.py (hunk 2 above)
Comment (maartenbreddels) [reply packages/vaex-core/vaex/dataframe.py]: Note that the unittest will catch it, but it's good to have this in the code.

File: tests/sort_test.py [hunk 4]
```diff
@@ -26,3 +26,19 @@ def test_sort_filtered():
...
+    assert df_sorted_1.z.tolist() == ['dog', 'cat', 'cat', 'dog', 'mouse']
+
+    # Case 2: str key followed by numeric key
+    df_sorted_2 = df.sort(by=['z', 'x'])
+    assert df_sorted_2.values.tolist() == [[1, 4], [3, 3], [1, 2], [5, 0], [5, 1]]
+    assert df_sorted_2.z.tolist() == ['cat', 'cat', 'dog', 'dog', 'mouse']
```
Comment (maartenbreddels) [inline tests/sort_test.py]: cool!

**Comment by JovanVeljanoski** [comment]
Thanks for the comments @maartenbreddels, useful stuff. I will get it sorted out soon.

File: packages/vaex-core/vaex/dataframe.py (hunk 1 above)
Comment (JovanVeljanoski) [reply packages/vaex-core/vaex/dataframe.py]: Good idea - implemented.

File: packages/vaex-core/vaex/dataframe.py (hunk 2 above)
Comment (JovanVeljanoski) [reply packages/vaex-core/vaex/dataframe.py]: Fair point - comment added.

File: tests/sort_test.py (hunk 3 above)
Comment (JovanVeljanoski) [reply tests/sort_test.py]: Ah..i keep on forgetting that. Now there is an assertion for each column.

**Pull request merged**